subsys/bluetooth/host/iso.c

```diff
@@ -705,29 +705,45 @@ static int bt_iso_setup_data_path(struct bt_conn *conn)
 		.dir = BT_HCI_DATAPATH_DIR_CTLR_TO_HOST };
 	struct bt_iso_data_path in_path = {
 		.dir = BT_HCI_DATAPATH_DIR_HOST_TO_CTLR };
+	struct bt_iso_chan_io_qos *tx_qos;
+	struct bt_iso_chan_io_qos *rx_qos;
 
 	chan = SYS_SLIST_PEEK_HEAD_CONTAINER(&conn->channels, chan, node);
 	if (!chan) {
 		return -EINVAL;
 	}
 
-	in_path.path = chan->qos->tx->path ? chan->qos->tx->path : &path;
-	out_path.path = chan->qos->rx->path ? chan->qos->rx->path : &path;
+	tx_qos = chan->qos->tx;
+	rx_qos = chan->qos->rx;
 
-	if (!chan->qos->tx) {
+	in_path.path = tx_qos && tx_qos->path ? tx_qos->path : &path;
```

Comment:
Can the path variable be renaming to something that is easier to understand?
disabled_path perhaps?

Looks like initializing it to BT_ISO_DATA_PATH_DISABLED could make some of the code below simpler as well.

Reply:
If I understand it correctly, then the `path` variable isn't a diabled path, but rather a default (HCI) path. 

There are some parts of this that I'm not 100% sure about, so we'd have to get @Vudentz input, but will at least rename `path` to `default_path`.

Reply:
But if tx_qos is NULL then it is set to PATH_DISABLED below, I'm not sure what it is set to if rx_qos->path is NULL though.

Reply:
It's confusing to me as well as both `struct bt_iso_data_path` and `struct bt_iso_chan_path` has a path ID (pid). @Vudentz Do you have any input here?